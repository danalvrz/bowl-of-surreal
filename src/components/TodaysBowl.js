import React from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Loading from '../pages/LoadingScreen';
import ErrorScreen from '../pages/ErrorScreen';

const TodaysBowl = (payload) => {
  const responseStatus = useSelector((state) => state.responseStatus);
  const artArray = [];
  let bgColor = '';
  const { visibleArtworkList } = payload;
  if (payload !== undefined && visibleArtworkList.length > 0) {
    visibleArtworkList.forEach((artwork) => {
      if (artwork.color !== null) {
        bgColor = `hsl(${artwork.color.h}, ${artwork.color.s}%, ${artwork.color.l}%)`;
      } else {
        bgColor = 'hsl(40, 2%, 21%)';
      }
      artArray.push(
        <Link
          to={`/artwork/${artwork.id}`}
          key={artwork.id}
          style={{ backgroundColor: bgColor }}
          className="
      era-card flex h-48
      w-1/2 lg:w-1/4 text-white"
        >
          <div className="h-full w-full flex-row p-5">
            <p
              className={`font-Work mb-auto text-right text-lg font-bold ${
                artwork.color.l > 80 ? 'text-neutral-400' : 'text-white'
              }`}
            >
              {artwork.title.length > 60
                ? artwork.title.substr(0, 60).concat('...')
                : artwork.title}
            </p>
            <p
              className={`font-Lato mt-1 text-right text-sm font-bold ${
                artwork.color.l > 80 ? 'text-neutral-400' : 'text-white'
              }`}
            >
              {artwork.artist_title ? artwork.artist_title : 'Unknown'}
            </p>
          </div>
        </Link>,
      );
    });
    return artArray;
  }
  if (responseStatus === 'LOADING') {
    return <Loading />;
  }
  if (responseStatus === 'ERROR') {
    return <ErrorScreen />;
  }
  return (
    <div className="font-Lato mx-auto h-screen p-20 text-center text-3xl font-bold text-white">
      Oops! Seems like today&lsquo;s bowl doesn&lsquo;t have that color!
      {' '}
      <br />
      {' '}
      <br />
      {' '}
      Try a different one..
    </div>
  );
};
export default TodaysBowl;
