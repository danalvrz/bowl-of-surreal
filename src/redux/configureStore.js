// eslint-disable-next-line import/no-extraneous-dependencies
import { createStore, applyMiddleware, compose } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from './rootReducer';

const composedEnhancer = compose(applyMiddleware(logger), applyMiddleware(thunk));
const store = createStore(
  rootReducer,
  composedEnhancer,
);

export default store;
