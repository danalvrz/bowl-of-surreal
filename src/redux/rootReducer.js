// eslint-disable-next-line import/no-extraneous-dependencies
import { combineReducers } from 'redux';
import artworks from './artworks/artworks';
import colorsFilter from './colors/colorsFilter';
import responseStatus from './status/status';
import optionsStatus from './options/options';

const rootReducer = combineReducers({
  artworks,
  colorsFilter,
  responseStatus,
  optionsStatus,
});

export default rootReducer;
