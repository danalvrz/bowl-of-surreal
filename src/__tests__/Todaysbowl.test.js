import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import TodaysBowl from '../components/TodaysBowl';

const mockArtworksReducer = () => [{
  id: 0, title: 'test', date_display: 'test', place_of_origin: 'test', image_id: 0, medium_display: 'test',
}];
const mockColorsReducer = () => [{
  id: 0, title: 'test', date_display: 'test', place_of_origin: 'test', image_id: 0, medium_display: 'test',
}];

const mockReducer = () => combineReducers({
  mockArtworksReducer,
  mockColorsReducer,
});
const store = createStore(mockReducer);

describe('Todays bowl section', () => {
  it('renders TodaysBowl correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <TodaysBowl visibleArtworkList={mockColorsReducer} />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
