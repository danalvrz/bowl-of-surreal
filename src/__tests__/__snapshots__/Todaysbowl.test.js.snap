// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todays bowl section renders TodaysBowl correctly 1`] = `
<div
  className="text-white font-Lato text-3xl font-bold mx-auto p-20 h-screen text-center"
>
  Oops! Seems like today‘s bowl doesn‘t have that color!
   
  <br />
   
   
  <br />
   
  Try a different one..
</div>
`;
